kernel/src/x86_64/arch_api/paging.rs

```diff
@@ -30,42 +30,73 @@ struct PageTableEntry {
     no_execute: bool,
 }
 
+const PRESENT_BIT: u64 = 1;
+const WRITEABLE_BIT: u64 = 1 << 1;
+const USER_ACCESSIBLE_BIT: u64 = 1 << 2;
+const WRITE_THROUGH_BIT: u64 = 1 << 3;
+const CACHE_DISABLED_BIT: u64 = 1 << 4;
+const ACCESSED_BIT: u64 = 1 << 5;
+const DIRTY_BIT: u64 = 1 << 6;
+const HUGE_PAGE_BIT: u64 = 1 << 7;
+const GLOBAL_BIT: u64 = 1 << 8;
+const NO_EXECUTE_BIT: u64 = 1 << 63;
+
+const PHYSICAL_ADDRESS_MASK: u64 = 0x000f_ffff_ffff_f000;
+
+const VIRTUAL_ADDRESS_SIGN_BIT: usize = 1 << 47;
```

Comment:
I like the use of constants a lot. The re-use of these bit masks is really awesome. Imagine if there was a mistake in one of them, and you fixed it only in the construction of the page table and not the deconstruction.